src/components/algorithm/EditAlgorithm.js

```diff
@@ -0,0 +1,232 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { withTranslation } from 'react-i18next';
+import { connect } from 'react-redux';
+import { withStyles } from '@material-ui/core/styles';
+import { withRouter } from 'react-router';
+import { Map } from 'immutable';
+
+import TextField from '@material-ui/core/TextField';
+import Container from '@material-ui/core/Container';
+import Alert from '@material-ui/lab/Alert';
+import Grid from '@material-ui/core/Grid';
+import Button from '@material-ui/core/Button';
+import SaveIcon from '@material-ui/icons/Save';
+
+import Editor from '../common/Editor';
+import Loader from '../common/Loader';
+import Main from '../common/Main';
+import { addAlgorithm, getAlgorithm, saveAlgorithm } from '../../actions';
+import {
+  AUTHOR_GRAASP,
+  AUTHOR_USER,
+  EDITOR_PROGRAMMING_LANGUAGES,
+} from '../../config/constants';
+import BackButton from '../common/BackButton';
+
+const styles = (theme) => ({
+  button: {
+    margin: theme.spacing(1),
+  },
+  infoAlert: {
+    marginTop: theme.spacing(2),
+    marginBottom: theme.spacing(2),
+  },
+});
+
+class EditAlgorithm extends Component {
+  state = {
+    name: '',
+    description: '',
+    code: '',
+  };
+
+  static propTypes = {
+    algorithm: PropTypes.instanceOf(Map).isRequired,
+    match: PropTypes.shape({
+      params: PropTypes.shape({
+        id: PropTypes.string,
+      }).isRequired,
+    }).isRequired,
+    history: PropTypes.shape({
+      goBack: PropTypes.func.isRequired,
+    }).isRequired,
+    dispatchGetAlgorithm: PropTypes.func.isRequired,
+    dispatchSaveAlgorithm: PropTypes.func.isRequired,
+    dispatchAddAlgorithm: PropTypes.func.isRequired,
+    classes: PropTypes.shape({
+      button: PropTypes.string.isRequired,
+      infoAlert: PropTypes.string.isRequired,
+    }).isRequired,
+    t: PropTypes.func.isRequired,
+    activity: PropTypes.bool.isRequired,
+  };
+
+  componentDidMount() {
+    const {
+      dispatchGetAlgorithm,
+      match: {
+        params: { id },
+      },
+    } = this.props;
+
+    dispatchGetAlgorithm({ id });
+  }
+
+  componentDidUpdate({ algorithm: prevAlgorithm }) {
+    const { algorithm } = this.props;
+    if (algorithm && algorithm !== prevAlgorithm) {
+      // eslint-disable-next-line react/no-did-update-set-state
+      this.setState({
+        name: algorithm.get('name'),
+        description: algorithm.get('description'),
+        code: algorithm.get('code'),
+      });
+    }
+  }
+
```

Comment:
Add `componentWillUnMount` to clear the state (remove the current algorithm data) to go along with the issue #66